xdsl/interactive/app.py

```diff
@@ -79,23 +89,38 @@ def compose(self) -> ComposeResult:
                     yield Button("Copy Output", id="copy_output_button")
         yield Footer()
 
+    @on(SelectionList.SelectedChanged)
+    def update_selected_view(self) -> None:
+        """
+        When the SelectionList (pass options) changes (i.e. a pass was selected or deselected), update the label to show
+        the query, and then call the update_current_module() function, which applies the selected passes to the input and displays the output
+        """
+        self.update_current_module()
+
     @on(TextArea.Changed, "#input")
     def update_current_module(self) -> None:
         """
-        Function called when the Input TextArea is cahnged. This function parses the Input IR and updates
-        the current_module reactive variable.
+        Function called when the Input TextArea is changed or a pass is selected/
+        unselected. This function parses the Input IR, applies selected passes and
+        updates
         """
         input_text = self.input_text_area.text
+        selected_passes = self.passes_selection_list.selected
+
         try:
             ctx = MLContext(True)
             for dialect in get_all_dialects():
                 ctx.load_dialect(dialect)
             parser = Parser(ctx, input_text)
             module = parser.parse_module()
+            pipeline = PipelinePass([p() for p in selected_passes])
+            pipeline.apply(ctx, module)
             self.current_module = module
         except Exception as e:
             self.current_module = e
 
+        self.watch_current_module()
```

Comment:
this isn't right, the watch_current_module should be called automatically when it's updated, if it isn't then we have a problem that we should fix